pygen/pygen_src/isa/riscv_instr.py

```diff
@@ -264,120 +261,187 @@ def set_imm_len(self):
                 self.imm_len = 5
             else:
                 self.imm_len = 11
-        self.imm_mask = self.imm_mask << self.imm_len
+        self.imm_mask = (self.imm_mask << self.imm_len) & self.shift_t
 
     def extend_imm(self):
         sign = 0
-        self.imm = self.imm << (32 - self.imm_len)
-        # sign = imm[31]
-        sign = self.imm.bin[0:1:1]
-        self.imm = self.imm >> (32 - self.imm_len)
+        # self.shift_t = 2 ** 32 -1 is used to limit the width after shift operation
+        self.imm = self.imm << (32 - self.imm_len) & self.shift_t
+        sign = (self.imm & 0x80000000) >> 31
+        self.imm = self.imm >> (32 - self.imm_len) & self.shift_t
         # Signed extension
         if((sign and not(self.format == "U_FORMAT")) or (self.imm_type in ["UIMM", "NZUIMM"])):
             self.imm = self.imm_mask | self.imm
 
     def post_randomize(self):
+        pass
+
+    # TODO : This needs to be fix later
+    # Using a explicit post_process function calling it from riscv_instr_stream class
+    # because for some cases the post_randomization was not properly called
+    # and getting the output of imm value as none
+    def post_process(self):
         self.extend_imm()
         self.update_imm_str()
 
-    def convert2asm(self):
-        pass
+    def convert2asm(self, prefix=" "):
+        asm_str = pkg_ins.format_string(string = self.get_instr_name(),
+                                        length = pkg_ins.MAX_INSTR_STR_LEN)
+        if(self.category.name != "SYSTEM"):
+            if self.format.name == "J_FORMAT":
+                asm_str = '{} {}, {}'.format(asm_str, self.rd.name, self.get_imm())
+            elif self.format.name == "U_FORMAT":
+                asm_str = '{} {}, {}'.format(asm_str, self.rd.name, self.get_imm())
+            elif self.format.name == "I_FORMAT":
+                if(self.instr_name == "NOP"):
+                    asm_str = "nop"
+                elif(self.instr_name == "WFI"):
+                    asm_str = "wfi"
+                elif(self.instr_name == "FENCE"):
+                    asm_str = "fence"
+                elif(self.instr_name == "FENCE_I"):
+                    asm_str = "fence.i"
+                elif(self.category.name == "LOAD"):
+                    asm_str = '{} {}, {},({})'.format(
```

Comment:
nit: I don't think there should be a comma between `{}` and `({})`, here and for the `STORE` instructions below